nengo/builder/tests/test_signal.py

```diff
@@ -123,6 +124,39 @@ def test_signal_reshape():
     assert three_d.reshape((2, -1, 2)).shape == (2, 2, 2)
     assert three_d.reshape((1, 2, 1, 2, 2, 1)).shape == (1, 2, 1, 2, 2, 1)
 
+    # check with non-contiguous arrays (and with offset)
+    value = np.arange(20).reshape(5, 4)
+    s = Signal(np.array(value), name='s')
+
+    s0slice = slice(0, 3), slice(None, None, 2)
+    s0shape = 2, 3
+    s0 = s[s0slice].reshape(*s0shape)
+    assert s.offset == 0
+    assert np.array_equal(s0.initial_value, value[s0slice].reshape(*s0shape))
+
+    s1slice = slice(1, None), slice(None, None, 2)
+    s1shape = 2, 4
+    s1 = s[s1slice].reshape(s1shape)
+    assert s1.offset > 0
```

Comment:
Don't we know exactly what the offset should be here?

Reply:
Yeah, you could figure it out. I just wrote it generally in case someone changes the test. The main thing I wanted to check is that we're testing one case with zero offset and one with non-zero offset, so that's why I put the asserts in. The point of this test is not to check the offset (there's a separate test for that), but there's also no reason we can't check the exact offset here.